library/Zend/Di/InstanceManager.php

```diff
@@ -502,7 +502,12 @@ protected function createHashForValues($classOrAlias, $paramValues)
                     $hashValue .= $param . '|';
                     break;
                 case 'array':
-                    $hashValue .= 'Array|';
+                    try {
+                        $hashValue .= serialize($param) . '|';
+                    } catch (\Exception $ex) {
```

Comment:
@weierophinney 
Thank you.
I got it.

Even though the serialization with suppressing Exception is not perfect, I had thought it is a pragmatic solution because optimistic hash is effective in most cases.

By the way, 
 If it is important to deal with both E_NOTICE and Exception,
I think var_export is more simple to get a hash of array than serialize.
Is it ok to use var_export?